### PR TITLE
Push the FS 77 Changes related to the complex queries

### PR DESCRIPTION
**Pull request by gaganahluwalia**
## Description
These changes are for the US - FS77, where complex queries were not working

## Changelog
- Major changes were done in the way Cypher Query is generated along with the way an answer is validated.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -49,15 +49,15 @@ FILE_AGENT_LLM="openai"
...
-SUGGESTIONS_MODEL="gpt-4o mini"
-REDIS_HOST="redis"
+DATASTORE_AGENT_MODEL="gpt-4o-mini"
+MATHS_AGENT_MODEL="gpt-4o-mini"
+WEB_AGENT_MODEL="gpt-4o-mini"
+CHART_GENERATOR_MODEL="gpt-4o-mini"
+ROUTER_MODEL="gpt-4o-mini"
+FILE_AGENT_MODEL="gpt-4o-mini"
+SUGGESTIONS_MODEL="gpt-4o-mini"
+REDIS_HOST="localhost"
```
Comment (IMladjenovic) [inline .env.example:62]: Can we move REDIS condig somewhere else in this file? I don't think it fits under `# model`

Comment (gaganahluwalia) [reply .env.example]: It's done, I have moved it up.

File: backend/src/prompts/templates/generate-cypher-query.j2 [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
+    
+6. **Property Matching**: Adhere to exact property values and capitalization in the schema (e.g., 'Aviation' and 'Social').
 
-For example, an expense of -45 is greater than an expense of -15.
+7. **Single Result for Maximum/Minimum**:
+    - For queries seeking a single result with the "highest" or "lowest" value, use `ORDER BY` and `LIMIT 1` to return only the top result.
+    - Example: If finding the fund with the highest ESG social score, sort by `esg.Score DESC` and limit to 1 result.
 
-When returning a value, always remove the `-` sign before the number.
+8. **Expense Handling**:
```
Comment (mic-smith) [inline backend/src/prompts/templates/generate-cypher-query.j2:45]: Do we need still need this expense handling logic?

Comment (gaganahluwalia) [reply backend/src/prompts/templates/generate-cypher-query.j2:45]: We don't actually need this. I just labelled the whole validation prompt. There is a lot of legacy code related to InferGPT, so I believe we should create a Ticket later on to remove all that.

**Review by mic-smith — APPROVED** [review]

**Review by IMladjenovic — APPROVED** [review]

**Pull request merged**